### ISSUE TITLE
Platform library does not build, when used inside the project

### ISSUE THREAD

**Issue by fkolar** [SAP/fundamental-ngx#3772]
#### Is this a bug, enhancement, or feature request?
Bug

#### Briefly describe your proposal.
When you add `@fundamental-ngx/platform` to your project where you have `lib/ `and` app/ `like structure and you add `FdpFormGroupModule` import ot your` library module`, the library does not build with error from platform. 

<img width="1300" alt="Screen Shot 2020-11-06 at 10 59 59 AM" src="https://user-images.githubusercontent.com/17149942/98353190-47772e00-201f-11eb-94ab-da3fa3851bcb.png">

#### Which versions of Angular and Fundamental Library for Angular are affected? (If this is a feature request, use current version.)

```
Angular CLI: 10.0.8
Node: 12.0.0
OS: darwin x64

Angular: 10.0.14
... animations, common, compiler, compiler-cli, core, forms
... platform-browser, platform-browser-dynamic, router
Ivy Workspace: Yes

Package                            Version
------------------------------------------------------------
@angular-devkit/architect          0.1000.8
@angular-devkit/build-angular      0.1000.8
@angular-devkit/build-ng-packagr   0.1000.8
@angular-devkit/build-optimizer    0.1000.8
@angular-devkit/build-webpack      0.1000.8
@angular-devkit/core               10.0.8
@angular-devkit/schematics         10.0.8
@angular/cdk                       10.2.7
@angular/cli                       10.0.8
@angular/language-service          10.2.2
@angular/localize                  10.2.2
@ngtools/webpack                   10.0.8
@schematics/angular                10.0.8
@schematics/update                 0.1000.8
ng-packagr                         10.1.2
rxjs                               6.5.5
typescript                         3.9.5
webpack                            4.43.0
"@fundamental-ngx/core": "0.23.0",
    "@fundamental-ngx/platform": "0.23.0",

```

#### If this is a bug, please provide steps for reproducing it.

https://github.com/fkolar/fundamental-error

or steps:

```
ng new fundamental-error --style=scss
cd test-fundamental-platform

ng add @nrwl/workspace
ng add @nrwl/angular 

 nx generate library myLib --publishable --importPath="@myorg/my-lib"


ng add @fundamental-ngx/core   (say no to everything )
ng add @fundamental-ngx/platform   (say no to everything )


ng build my-lib
```

**Comment by mykolavorobiov** [comment]
@fkolar Could you please pre-setup the project with your issue. I can't reproduce your case. (MacOS)
![image](https://user-images.githubusercontent.com/38053194/98549196-232b8380-22a3-11eb-87d6-3ca12fa982a4.png)
![image](https://user-images.githubusercontent.com/38053194/98549135-0ee78680-22a3-11eb-8bb9-f104aa26702c.png)

**Comment by Betrozov** [comment]
the same result (Linux)
![angular-v](https://user-images.githubusercontent.com/8318592/98552673-6be53b80-22a7-11eb-9b16-9e1151643427.png)
![build](https://user-images.githubusercontent.com/8318592/98552694-6ee02c00-22a7-11eb-8d84-d0aac8e82745.png)

**Comment by mykolavorobiov** [comment]
@fkolar 
```ng new my-app --style=scss
cd my-app
reinstall typescript with version 3.9.7 (it doesn't work with v.4 and high)
ng add @nrwl/workspace
nx add @fundamental-ngx/core
nx add @fundamental-ngx/platform
npm install @angular/cdk
npm install hammerjs

rename tsconfig.json -> tsconfig.base.json
nx generate library myLib --publishable --importPath="@myorg/my-lib"

add platform module to your lib
Like this
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FdpFormGroupModule } from '@fundamental-ngx/platform';
@NgModule({
  imports: [CommonModule, FdpFormGroupModule],
})
export class MyLibModule {}

nx build lib
```
![image](https://user-images.githubusercontent.com/38053194/98669465-17e75f00-235a-11eb-9d83-852c324e7725.png)

**Comment by mykolavorobiov** [comment]
![image](https://user-images.githubusercontent.com/38053194/98670937-220a5d00-235c-11eb-8f2d-66ac0e95a562.png)

**Comment by fkolar** [comment]
I tried to re-run the steps, there is missing adding @nrwl/angular library but It works, then I compared it againts settings on the other project and here are the differences. 

https://github.com/fkolar/fundamentals-error/commit/1eabd3a5dfbcc663f7511021eae2e1c175cf7c1d

**Comment by fkolar** [comment]
I found real problem behind this. If you add this to your compiler options it will fails

"angularCompilerOptions": {
    "fullTemplateTypeCheck": true,

Does this mean that we have some problem inside the template ?

**Comment by dimamarksman** [comment]
> I found real problem behind this. If you add this to your compiler options it will fails
> 
> "angularCompilerOptions": {
> "fullTemplateTypeCheck": true,
> 
> Does this mean that we have some problem inside the template ?

Looks like that

**Comment by dimamarksman** [comment]
The same error we can see running `ng xi18n` so this might be related to the https://github.com/SAP/fundamental-ngx/issues/4846

**Comment by kavya-b** [comment]
@fkolar @manjunathanagaraj closing this issue as it seems like this was fixed with #4889 .
please reopen if this issue still persists.
